crates/allocator/src/bump.rs

```diff
@@ -137,9 +137,31 @@ impl InnerAlloc {
     }
 }
 
+#[inline]
+fn required_pages(size: usize) -> Option<usize> {
+    size.checked_add(PAGE_SIZE - 1)
+        .and_then(|num| num.checked_div(PAGE_SIZE))
+}
+
 #[cfg(test)]
 mod tests {
     use super::*;
+    use std::mem::size_of;
+
+    #[test]
+    fn can_alloc_no_bytes() {
+        let mut inner = InnerAlloc::new();
+
+        let layout = Layout::new::<()>();
+        assert_eq!(inner.alloc(layout), Some(0));
+
+        let expected_limit =
+            PAGE_SIZE * required_pages(layout.pad_to_align().size()).unwrap();
```

Comment:
I'm gonna push back a bit on this, since most of the `unwrap()` come after calls to `required_pages()`. In the tests we have preconditions on the size of allocations, ensuring that no individual allocation is going to overflow (unless the test checks for overflows, in which case we don't request pages). This means getting the required number of pages is safe. Adding `expect()`'s and messages would add unnecessary noise to the tests imo.

That said, if there are specific places where you think think we'd benefit from `expect()`, feel free to point them out.